packages/tooling/fast-tooling-wasm/permutator/permutate_number.c

```diff
@@ -14,21 +16,41 @@ double get_random_number(cJSON *configSchema)
     cJSON *maximum = cJSON_GetObjectItemCaseSensitive(configSchema, "maximum");
     cJSON *exclusiveMaximum = cJSON_GetObjectItemCaseSensitive(configSchema, "exclusiveMaximum");
 
-    const int multipleOfValue = multipleOf->valueint
-        ? multipleOf->valueint
+    const bool minimumDefined = (minimum && minimum->valuedouble) || (exclusiveMinimum && exclusiveMinimum->valuedouble);
+
+    const bool maximumDefined = (maximum && maximum->valuedouble) || (exclusiveMaximum && exclusiveMaximum->valuedouble);
+
+    const double multipleOfValue = multipleOf && multipleOf->valuedouble
+        ? multipleOf->valuedouble
         : 1;
-    const int minimumValue = minimum->valueint
-        ? minimum->valueint
-        : exclusiveMinimum->valueint
-        ? exclusiveMinimum->valueint + 1
+    double minimumValue = minimum && minimum->valuedouble
```

Comment:
In some cases you are camelCasing and in others, you are not?  Which pattern should be followed and can we make consistent? for example, "minimumValue" and "valuedouble"